### PR TITLE
feat(foundation): implement `getExternalUrl` function

### PR DESCRIPTION
**Pull request by mmackz**
- implement `getExternalUrl` function
- add tests

- [x] https://github.com/rabbitholegg/questdk-plugins/pull/478

BOOST-4268

https://linear.app/rh-app/issue/BOOST-4268/implement-getexternalurl-in-plugins-foundation

### CODE REVIEW

File: packages/foundation/src/Foundation.test.ts [hunk 1]
```diff
@@ -403,4 +403,37 @@ describe('Given the foundation plugin', () => {
...
+    })
+
+    test('should return the base url for 1155 mint', async () => {
+      const mint: MintActionParams = {
+        chainId: Chains.BASE,
+        contractAddress: '0x1d2550d198197df1a10af515cf2ea0d790889b93',
+        tokenId: 213,
+      }
+      const result = await getExternalUrl(mint)
+      expect(result).toBe('https://foundation.app/')
```
Comment (jonathandiep) [inline packages/foundation/src/Foundation.test.ts:426]: This test result doesn't look correct? Shouldn't it also include the action params in here?

Comment (jonathandiep) [reply packages/foundation/src/Foundation.test.ts:426]: Nvm I see that this is intentional. Is it possible to extract the url from Foundation? Or will it be a super deep dive into their API or something?

**Review by jonathandiep — APPROVED** [review]
lgtm, but would like to try to get the 1155 external url (if it's low lift!)

File: packages/foundation/src/Foundation.test.ts (hunk 1 above)
Comment (mmackz) [reply packages/foundation/src/Foundation.test.ts:426]: There erc-1155 implementation is custom and has no available frontend. The last one was for their babes collection.

https://babes.foundation.app/

Comment (mmackz) [reply packages/foundation/src/Foundation.test.ts:426]: It might be available in the contract or token uri

Comment (mmackz) [reply packages/foundation/src/Foundation.test.ts:426]: No luck.

**Comment by mmackz** [comment]
> lgtm, but would like to try to get the 1155 external url (if it's low lift!)

Was not able to find any way to derive an external url for 1155 unfortunately. Will update if something comes up.

**Pull request merged**